src/can.rs

```diff
@@ -758,6 +758,36 @@ impl Frame {
         }
     }
 
+    pub fn new_self(id: u32, extended: bool, data: &[u8]) -> Option<Self> {
```

Comment:
I am not to deep into CAN itself, though from an API perspective i am wondering, we would now create 3 different new methods that have basically the same structure and just depending on the "extended" bool would set a different value. Why do we not combine them into one method that takes a enum representing all 3 cases?